### PR TITLE
Make default StaticRootPath compile time settable

### PR DESCRIPTION
**Pull request by 6543**
be able to compile default location of StaticRootPath independent from AppWorkPath

so you can compile with
```sh
	LDFLAGS="-X \"code.gitea.io/gitea/modules/setting.StaticRootPath=/usr/share/webapps/gitea/\""
	LDFLAGS="$LDFLAGS -X \"code.gitea.io/gitea/modules/setting.AppWorkPath=/var/lib/gitea/\""
	export LDFLAGS
```

### CODE REVIEW

File: modules/setting/setting.go [hunk 1]
```diff
@@ -66,15 +66,16 @@ const (
...
+	AppVer            string
+	AppBuiltWith      string
+	AppName           string
+	AppURL            string
+	AppSubURL         string
+	AppSubURLDepth    int // Number of slashes
+	AppPath           string
+	AppDataPath       string
+	AppWorkPath       string
+	AppStaticRootPath string
```
Comment (zeripath) [inline modules/setting/setting.go]:
```suggestion
	AppVer         string
	AppBuiltWith   string
	AppName        string
	AppURL         string
	AppSubURL      string
	AppSubURLDepth int // Number of slashes
	AppPath        string
	AppDataPath    string
	AppWorkPath    string
```

File: modules/setting/setting.go [hunk 2]
```diff
@@ -652,7 +653,10 @@ func NewContext() {
 	PortToRedirect = sec.Key("PORT_TO_REDIRECT").MustString("80")
 	OfflineMode = sec.Key("OFFLINE_MODE").MustBool()
 	DisableRouterLog = sec.Key("DISABLE_ROUTER_LOG").MustBool()
-	StaticRootPath = sec.Key("STATIC_ROOT_PATH").MustString(AppWorkPath)
+	if len(AppStaticRootPath) == 0 {
+		AppStaticRootPath = AppWorkPath
+	}
+	StaticRootPath = sec.Key("STATIC_ROOT_PATH").MustString(AppStaticRootPath)
```
Comment (zeripath) [inline modules/setting/setting.go]:
```suggestion
	if len(StaticRootPath) == 0 {
		StaticRootPath = AppWorkPath
	}
	StaticRootPath = sec.Key("STATIC_ROOT_PATH").MustString(StaticRootPath)
```

**Review by zeripath — COMMENTED** [review]
I don't quite understand the need for the new AppStaticRootPath variable - it's not as if all the App prefixed variables are compile time settable.

**Comment by zeripath** [comment]
Need to also update `installation/from-source.en-us.md`:

```patch
diff --git a/docs/content/doc/installation/from-source.en-us.md b/docs/content/doc/installation/from-source.en-us.md
index cf3f599c5..9c7016577 100644
--- a/docs/content/doc/installation/from-source.en-us.md
+++ b/docs/content/doc/installation/from-source.en-us.md
@@ -143,7 +143,7 @@ Gitea will search for a number of things from the `CustomPath`. By default this
 the `custom/` directory in the current working directory when running Gitea. It will also
 look for its configuration file `CustomConf` in `$CustomPath/conf/app.ini`, and will use the
 current working directory as the relative base path `AppWorkPath` for a number configurable
-values.
+values. Finally the static files will be served from `StaticRootPath` which defaults to the `AppWorkPath`.
 
 These values, although useful when developing, may conflict with downstream users preferences.
 
@@ -154,6 +154,7 @@ using the `LDFLAGS` environment variable for `make`. The appropriate settings ar
 * To set the `CustomPath` use `LDFLAGS="-X \"code.gitea.io/gitea/modules/setting.CustomPath=custom-path\""`
 * For `CustomConf` you should use `-X \"code.gitea.io/gitea/modules/setting.CustomConf=conf.ini\"`
 * For `AppWorkPath` you should use `-X \"code.gitea.io/gitea/modules/setting.AppWorkPath=working-path\"`
+* For `StaticRootPath` you should use `-X \"code.gitea.io/gitea/modules/setting.StaticRootPath=static-root-path\"`
 
 Add as many of the strings with their preceding `-X` to the `LDFLAGS` variable and run `make build`
 with the appropriate `TAGS` as above.
```

**Comment by zeripath** [comment]
@6543 I hope you don't mind that I've just made those above changes.

File: modules/setting/setting.go [hunk 3]
```diff
@@ -652,7 +652,10 @@ func NewContext() {
 	PortToRedirect = sec.Key("PORT_TO_REDIRECT").MustString("80")
 	OfflineMode = sec.Key("OFFLINE_MODE").MustBool()
 	DisableRouterLog = sec.Key("DISABLE_ROUTER_LOG").MustBool()
-	StaticRootPath = sec.Key("STATIC_ROOT_PATH").MustString(AppWorkPath)
+	if len(StaticRootPath) == 0 {
+		StaticRootPath = AppWorkPath
+	}
+	StaticRootPath = sec.Key("STATIC_ROOT_PATH").MustString(AppStaticRootPath)
```
Comment (zeripath) [inline modules/setting/setting.go]:
```suggestion
	StaticRootPath = sec.Key("STATIC_ROOT_PATH").MustString(StaticRootPath)
```

**Review by zeripath — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Comment by zeripath** [comment]
make lg-tm work

**Comment by 6543** [comment]
Thx

**Pull request merged**